build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/gem_builder.rb

```diff
@@ -37,6 +37,9 @@ def each(&block)
         code.spec_files.each do |path, code|
           y.yield("spec/#{path}", code)
         end
+        code.rbs_files.each do |path, code|
```

Comment:
I think you want to check `@service.support_rbs?` here, similar to `if @service.smoke_tests` above, since `rbs_files` method entirely relies on it being enabled.